src/nni_evaluation_engine.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019-2021 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+//
+// NNI Evaluation Engine
+// 
```

Comment:
General comment for discussion: we should continue to think about ways in which we can abstract away the evaluation of "goodness" of an NNI move beyond exactly GP.